modules/gateway/gateway.go

```diff
@@ -105,27 +133,34 @@ func (g *Gateway) Info() (info modules.GatewayInfo) {
 }
 
 // New returns an initialized Gateway.
-func New(tcps *network.TCPServer, s *consensus.State) (g *Gateway, err error) {
-	if tcps == nil {
-		err = errors.New("gateway cannot use nil tcp server")
-		return
-	}
+func New(addr string, s *consensus.State) (g *Gateway, err error) {
```

Comment:
I suppose. I feel like there was a good reason why we weren't doing that though...